### DISCUSSION TITLE
Follow the Leader Launcher Running Forever

### DISCUSSION THREAD

**Discussion by pralgomathic** [aws-deepracer/aws-deepracer-follow-the-leader-sample-project#6] [Q&A] [answered]
Hi, anybody of you worked on Follow the Leader project? While I was trying to load the required nodes to run it in the DeepRacer with  `ros2 launch ftl_launcher ftl_launcher.py ` but it seems running forever, I have been waiting for more than 12+ hours but yet still running. Any idea what's wrong there?

**Comment by pratik-nichat** [comment, marked as answer]
Hello Jumman,

Thank you for using the FTL sample project. Make sure you are on the latest software version for Ubuntu 20.04
  1.  You can follow the steps in [prerequisite](https://github.com/aws-deepracer/aws-deepracer-follow-the-leader-sample-project#prerequisites) section to [calibrate](https://docs.aws.amazon.com/deepracer/latest/developerguide/deepracer-calibrate-vehicle.html) your device. Once you have calibrated the device, follow the [instructions](https://github.com/aws-deepracer/aws-deepracer-follow-the-leader-sample-project#downloading-and-building) to download and build the FTL project. Make sure you stop the deepracer-core.service as mentioned in Step 2 of the [instructions](https://github.com/aws-deepracer/aws-deepracer-follow-the-leader-sample-project#downloading-and-building).
  2.  Once you are ready, [run](https://github.com/aws-deepracer/aws-deepracer-follow-the-leader-sample-project#running-the-node) the FTL launcher using your CLI.
  3.  To enable the car to move, using the CLI itself, enable the `followtheleader` mode following the instructions [here](https://github.com/aws-deepracer/aws-deepracer-follow-the-leader-sample-project#enabling-followtheleader-mode-using-the-cli).
  4.  IF you see "NO_ACTION" on the terminal, it means that the DeepRacer is not recognizing a person to follow in its field of view. On successful object detection, you should see the actions like FORWARD / REVERSE etc. on the terminal which means FTL launcher is working as expected.
  5.  If you still think the car is not moving, even after ensuring `followtheleader` mode is enabled and calibration is done as well, we also provide manually setting MAX_SPEED scale described [here](https://github.com/aws-deepracer/aws-deepracer-follow-the-leader-sample-project#changing-the-max_speed-scale-of-the-aws-deepracer). Note that due to battery level change, it may affect the default speed of the car, hence we provide the option to set MAX_SPEED which is very custom to each DeepRacer.
  6.  You can get more in-depth explanation of how the FTL project works and what is expected in the [getting started](https://github.com/aws-deepracer/aws-deepracer-follow-the-leader-sample-project/blob/main/getting-started.md) doc.
Let us know if you need any more help, thanks!

Best Regards,
Pratik Nichat
SDE DeepRacer

**Reply by pratik-nichat** [reply]
If you have more questions or want to discuss about your project or have any further issues, please email us at deepraceropensource@amazon.com along with some more details regarding your project and we will be happy to know more!

**Reply by pralgomathic** [reply]
Hi @pratik-nichat , Thank you for your detailed answers. While I still have some issues regarding moving the car smoothly, but I will follow the steps more carefully if in case I did any mistake in the calibration. Also, I need to discuss more regarding my planned research project. I'll be emailing to the given address,
